src/main/java/io/r2dbc/postgresql/PostgresqlConnectionFactory.java

```diff
@@ -224,15 +253,15 @@ public Mono<io.r2dbc.postgresql.api.PostgresqlReplicationConnection> replication
         return doCreateConnection(true, connectionStrategy).map(DefaultPostgresqlReplicationConnection::new);
     }
 
-    private Mono<PostgresqlConnection> doCreateConnection(UniformLoadBalancerConnectionStrategy connectionStrategy, boolean forReplication, ConnectionFunction connectionFunction, String host) {
+    private Mono<PostgresqlConnection> doCreateConnection(UniformLoadBalancerConnectionStrategy connectionStrategy, boolean forReplication, ConnectionFunction connectionFunction, String host, boolean isControlConnection) {
 
         ZoneId defaultZone = TimeZone.getDefault().toZoneId();
         SocketAddress endpoint = InetSocketAddress.createUnresolved(host, 5433);
```

Comment:
```suggestion
        SocketAddress endpoint = new InetSocketAddress(host, 5433);
```
Does this seem better since we want to use resolved addresses?

Reply:
The upstream code uses `InetSocketAddress.createUnresolved()` to create endpoints.